src/BuiltInTools/dotnet-watch/Program.cs

```diff
@@ -378,5 +382,20 @@ public void Dispose()
             _console.CancelKeyPress -= OnCancelKeyPress;
             _cts.Dispose();
         }
+
+        private static void RegisterAssemblyResolutionEvents()
+        {
+            var roslynPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..", "..", "..", "..", "..", "..", "Roslyn", "bincore");
+
+            AssemblyLoadContext.Default.Resolving += (context, assembly) =>
+            {
+                var assemblyFilename = $"{assembly.Name}.dll";
+                if (assemblyFilename == "Microsoft.CodeAnalysis.CSharp.dll" || assemblyFilename == "Microsoft.CodeAnalysis.dll")
```

Comment:
You could do the concat after you've checked the assembly name. Something like:

```C#
if (assembly.Name is "Microsoft.CodeAnalysis" or "Microsoft.CodeAnalysis.CSharp")
{
   return context.LoadFromAssemblyPath(Path.Combine(roslynPath, assemblyFilename + ".dll");
}
```

Avoids checking strings that you know would match.